src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -254,12 +268,16 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                     break;
 
                 case TableExpressionKind.Top:
-                    var sortOrder = context.GetFirstSortOrderForSupportedParam();
+                    var (paramInfo, sortOrder) = context.GetFirstSupportedSortParam();
+                    FindRestrictingPredecessorTableExpressionIndex();
```

Comment:
return value is discarded

Reply:
@johnstairs  I am not following...
Both of the return values are being used